test/asdf_nu.bats

```diff
@@ -110,3 +110,23 @@ cleaned_path() {
   result=$(echo "$output" | grep "ASDF INSTALLED PLUGINS:")
   [ "$result" != "" ]
 }
+
+@test "parses the output of asdf plugin list all" {
+  local expected="\
+bar                           http://example.com/bar
+dummy                        *http://example.com/dummy
+foo                           http://example.com/foo
+foos                          https://example.com/foo
+fooss                        *https://example.com/foo
+foog                          git@example.com/foo.git
+foogs                        *git@example.com/foo.git"
+  local nr_lines=7
+
+  run nu -c "
+    let re = '(?P<name>.+)\s+?(?P<installed>[*]?)(?P<repository>(?:git|http|https).+)'
```

Comment:
As with the test above, sourcing `asdf.nu` would then let you run the commands we're trying to assert. This way we would be testing the `asdf.nu` and not code duplicated from that script here.